examples/bluetooth/ble_remote_control/main/main.c

```diff
@@ -0,0 +1,252 @@
+            (buttons & 0x01) ? "A" : "-",
+            (buttons & 0x02) ? "B" : "-",
+            (buttons & 0x04) ? "C" : "-",
+            (buttons & 0x08) ? "D" : "-");
+    }
+
+    ESP_LOGI(HID_DEMO_TAG, " ----- End of user input ----- \n");
+}
+
+/**
+ * @brief   Main task to send user input to the ESP to the central device
+ *          
+ * @details This task will wait for user input from the console, joystick, or button (using RTOS queue)
+ *          The function does not return unless Tear Down button sequence is pressed (and tear down config enabled in menuconfig)
+*/
+void joystick_task() 
+{
+    esp_err_t ret = ESP_OK;
+    input_event_t input_event = {0};
+    uint8_t x_axis = 0;
+    uint8_t y_axis = 0;
+    uint8_t hat_switch = 0; // unused in this example
+    uint8_t button_in = 0;
+    uint8_t throttle = 0;   // unused in this example
+
+    while(true){
+        DELAY(HID_LATENCY);
+
+        if (!is_connected) {
+            ESP_LOGI(HID_DEMO_TAG, "Not connected, do not send user input yet");
+            DELAY(3000);
+            continue;
+        }
+
+        // HID report values to set is dependent on the HID report map (refer to hidd.c)
+        // For this examples, the values in the HID report to send are
+            // x_axis : 8 bit, 0 - 255
+            // y_axis : 8 bit, 0 - 255
+            // hat switches : 4 bit
+            // buttons 1 to 4: 1 bit each, 0 - 1
+            // throttle: 8 bit
+
+        // Send report if there are any inputs (refer to controller.c)
+        if (xQueueReceive(input_queue, &input_event, 100) == pdTRUE) {
+            switch (input_event.input_source) {
+            case (INPUT_SOURCE_BUTTON):
+                button_in = input_event.data_button;
+                joystick_adc_read(&x_axis, &y_axis);  
+                ESP_LOGI(HID_DEMO_TAG, "Button input received");
+                break;
+            case (INPUT_SOURCE_CONSOLE):
+                button_in = button_read();
+                char_to_joystick_input(input_event.data_console, &x_axis, &y_axis);
+                ESP_LOGI(HID_DEMO_TAG, "Console input received");
+                break;
+            case (INPUT_SOURCE_JOYSTICK):
+                button_in = button_read();
+                x_axis = input_event.data_joystick_x;
+                y_axis = input_event.data_joystick_y;
+                ESP_LOGI(HID_DEMO_TAG, "Joystick input received");
+                break;
+            default:
+                ESP_LOGE(HID_DEMO_TAG, "Unknown input source, source %d", input_event.input_source);
+                break;
+            }
+
+            set_hid_report_values(x_axis, y_axis, button_in, hat_switch, throttle);
+            print_user_input_report(x_axis, y_axis, hat_switch, button_in, throttle);
+            ret = send_user_input();
+        } 
+
+        // Alternatively, to simply poll user input can do:
+        // joystick_adc_read(&x_axis, &y_axis);
+        // button_read(&button_in);
```

Comment:
Are these comments still useful?

Reply:
The comments here are to provide a simplified implementation of the remote control (where user just poll for the inputs), if let's say the event queue implementation does not fit the user's use case.